handbook/engineering/web/testing.md

```diff
@@ -0,0 +1,177 @@
+# Testing web code
+
+This is the implementation of our [company testing philosophy](../testing.md) for all code (TypeScript, stylesheets, ...) used on the web, such as our webapp, browser extension, shared code, or Sourcegraph extensions.
+
+## General Goals
+
+1. **Our product and code works as intended when we ship it to customers.**
+
+2. **Our product and code doesn't accidentally break as we make changes over time.**
+
+3. **Failures should be clear about _what_ is not working and give as many hints as possible to _why_ it's not working.**
+
+4. **Tests should not commonly need manual updates when the implementation is changed.**
+   If tests commonly need manual updates, it creates the potential to accidentally change the test so that it doesn't catch a regression introduced in the same change. It also puts more burden on the code reviewers. In addition, updating tests can feel daunting and reduces the motivation to make changes to that part of the codebase.
+
+5. **Tests should be easy to write and maintain.**
```

Comment:
To be more consistent, we could pick between writing all of these goals with "should" or writing them as regular present tense like in 1 and 2